lib/rouge/lexers/ecl.rb

```diff
@@ -0,0 +1,140 @@
+        abs acos aggregate allnodes apply ascii asin asstring 
+        atan _token ave case catch choose choosen choosesets 
+        clustersize combine correlation cos cosh count covariance cron 
+        dataset dedup define denormalize dictionary distribute distributed distribution 
+        ebcdic enth error evaluate event eventextra eventname exists 
+        exp failcode failmessage fetch fromunicode fromxml getenv getisvalid 
+        global graph group hash hashcrc having httpcall httpheader 
+        if iff index intformat isvalid iterate join keyunicode 
+        length library limit ln local log loop map 
+        matched matchlength matchposition matchtext matchunicode max merge mergejoin 
+        min nofold nolocal nonempty normalize parse pipe power 
+        preload process project pull random range rank ranked 
+        realformat recordof regexfind regexreplace regroup rejected rollup round 
+        roundup row rowdiff sample set sin sinh sizeof 
+        soapcall sort sorted sqrt stepped stored sum table 
+        tan tanh thisnode topn tounicode toxml transfer transform 
+        trim truncate typeof ungroup unicodeorder variance which workunit 
+        xmldecode xmlencode xmltext xmlunicode apply assert build buildindex 
+        evaluate fail keydiff keypatch loadxml nothor notify output 
+        parallel sequential soapcall wait
+      )
+
+      class1 = %w(
+        file date str math metaphone metaphone3 uni audit blas system
+      )
+
+      class2 = %w(
+        debug email job log thorlib util workunit
+      )
+
+      state :string_character_escape do
+        rule %r/\\\\(x\\h{2}|[0-2][0-7]{,2}|3[0-6][0-7]?|37[0-7]?|[4-7][0-7]?|.|$)/, Text
+      end
+
+      state :single_quote do
+        rule %r([xDQUV]?'([^'\\]*(?:\\.[^'\\]*)*)'), Str::Single
+        mixin :string_character_escape
+      end
+
+      state :inline_whitespace do
+        rule %r/[ \t\r]+/, Text
+        rule %r/\\\n/, Text # line continuation
+        rule %r(/(\\\n)?[*].*?[*](\\\n)?/)m, Comment::Multiline
+      end
+
+      state :whitespace do
+        rule %r/\n+/m, Text
+        rule %r(//.*?$), Comment::Single
+        mixin :inline_whitespace
+      end
+
+      state :root do
+	mixin :whitespace
+	mixin :single_quote
+
+        rule %r(\b(?i:(#{functions.join('|')}))\s*(\()), Name::Function
+        rule %r(\b(?i:(#{keywords.join('|')}))\b), Keyword
+        rule %r(#(?i:(#{template.join('|')}))\b), Keyword::Declaration
+        rule %r(\b(?i:(std)\.(#{class1.join('|')})\.(#{class2.join('|')}))\b), Name::Class
+
+        rule %r(\b(?i:(#{typed.join('|')}))\d+\b), Keyword::Type
+        rule %r(\b(?i:(#{type.join('|')}))\b), Keyword::Type
+        rule %r(\b(?i:(integer|unsigned))[1-8]\b), Keyword::Type
+        rule %r(\b(?i:real)(4|8)\b), Keyword::Type
+        rule %r(\b(?i:(u)?decimal)(\d+(_\d+)?)\b), Keyword::Type
+
+        rule %r(\b(?i:(and|not|or|in))\b), Operator::Word
+        rule %r([:=||>|<|<>|/|\\|+|-|=]), Operator 
```

Comment:
Are the `|` characters supposed to be interspersed here? You don't need to do that with a character class. Please note because of the meaning `-` can have inside a character class, it should be the last character in a class if intended to match literally.

Reply:
There is one operator in ECL that is ":=".

Reply:
@dehilsterlexis Looks like you have and additional "|" at the start which is not needed:
```
  rule %r([:=||>|<|<>|/|\\|+|-|=]),
```
Should be:
```
  rule %r([:=|>|<|<>|/|\\|+|-|=]),
```
?

Reply:
Good catch.